src/main/java/com/example/weather/client/wrapper/HttpRequestWrapperImpl.java

```diff
@@ -0,0 +1,23 @@
+package com.example.weather.client.wrapper;
+
+import lombok.SneakyThrows;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.support.HttpRequestWrapper;
+
+import java.net.URI;
+
+public class HttpRequestWrapperImpl extends HttpRequestWrapper {
```

Comment:
```suggestion
public class ApiKeyHttpRequestWrapper extends HttpRequestWrapper {
```
`~Impl` suffix tells you only that it's implementation, but you can have multiple kinds of HttpRequestWrappers